### PR TITLE
removed selftest from Bio/KDTree/KDTree.py

### PR DESCRIPTION
**Pull request by Gasta88**
In reference to issue #820, I've removed the self test from this module. The test has been developed 6 months ago and sits correctly in `Tests/test_KDTree.py`

### CODE REVIEW

**Comment by peterjc** [comment]
Did you mean to include additions to ``Tests/test_KDTree.py`` as part of this pull request?

(I am not sure why AppVeyor is confused - GitHub says this branch can be merged?)

**Comment by Gasta88** [comment]
Yes, GitHub didn't display any problem.

I didn't add `Tests/test_KDTree.py` because it's already in place.

**Comment by peterjc** [comment]
The current ``Tests/test_KDTree.py`` does not seem to cover all the functionality used in the old self test. e.g. ``get_radii``

This is confirmed based on the current coverage reports - see https://codecov.io/gh/biopython/biopython/src/master/Bio/KDTree/KDTree.py

**Comment by Gasta88** [comment]
Thanks for pointing that out, I have one more query for this unittest.

for this module the creator wrote the testing methods inside the module itself, calling them later in `Tests/test_KDTree.py`.

Can I follow the same example or would it be best to move all the methods into the testing script?

**Comment by peterjc** [comment]
I would suggest moving those private test functions into the test suite (which can be done later, or as part of this pull request if you prefer).

File: Tests/test_KDTree.py [hunk 1]
```diff
@@ -12,31 +12,142 @@
     raise MissingExternalDependencyError(
         "Install NumPy if you want to use Bio.KDTree.")
 
-try:
```
Comment (peterjc) [inline Tests/test_KDTree.py]: Please lease the try/except - this is still used on Jython to skip this test.

**Comment by Gasta88** [comment]
The method `test_all_search` might raise few questions.

I've decided to test the `all_get_radii` and `all_get_indices` methods, but the computation is taking quite a toll on my machine. Therefore I've taken into account a subset of 100 iterations to test only.

The `test_KDTree_exceptions` is classic fuzzy testing to try few exceptions on the class.

**Comment by peterjc** [comment]
It seems the tests are timing out under PyPy (Python 2 and 3), can you try that locally? Perhaps even fewer iterations are needed.

**Comment by Gasta88** [comment]
Took me 20 seconds on my machine, what's the timeout limit for TravisCI?

**Comment by peterjc** [comment]
Its the time out for ``run_tests.py --offline`` we need to worry about.

**Comment by Gasta88** [comment]
Travis CI seems to have timed out for Python 3. 
I've checked with both versions (2.7 and 3) on my machine with the `run_tests.py --offline` and everything looks ok.

**Comment by peterjc** [comment]
I've had a play with the ``travis_wait`` command, e.g. https://travis-ci.org/peterjc/biopython/jobs/261067283 - not yet working as I'd like

It may be more practical to ask TravisCI nicely if they could increase the timeout for the biopython repository's builds...

**Comment by peterjc** [comment]
The ``travis_wait`` change in now included on the master branch via #1359 so that should help here with timeouts.

**Comment by peterjc** [comment]
It looks like there is a Python 3 specific problem:

```
======================================================================
ERROR: test_all_search (test_KDTree.KDTreeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_KDTree.py", line 199, in test_all_search
    self.assertTrue(test_all_search((nr_points / 10), dim, bucket_size, query_radius))
  File "/home/travis/build/biopython/biopython/Tests/test_KDTree.py", line 159, in test_all_search
    coords = random.random((nr_points, dim))
  File "mtrand.pyx", line 856, in mtrand.RandomState.random_sample (numpy/random/mtrand/mtrand.c:15541)
  File "mtrand.pyx", line 167, in mtrand.cont0_array (numpy/random/mtrand/mtrand.c:6141)
TypeError: 'float' object cannot be interpreted as an integer
----------------------------------------------------------------------
```

File: Tests/test_KDTree.py [hunk 2]
```diff
@@ -195,7 +195,8 @@ def test_KDTree(self):
             self.assertTrue(test(nr_points, dim, bucket_size, radius))
 
     def test_all_search(self):
-        self.assertTrue(test_all_search(nr_points, dim, bucket_size, query_radius))
+        for i in range(0, 5):
+            self.assertTrue(test_all_search((nr_points / 10), dim, bucket_size, query_radius))
```
Comment (peterjc) [inline Tests/test_KDTree.py]: Use explicit integer division here (``//`` not ``/``) in order to work as expected on Python 2 and 3

**Comment by peterjc** [comment]
Python 3.6 on Travis failed:

```
======================================================================
FAIL: test_all_search (test_KDTree.KDTreeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_KDTree.py", line 199, in test_all_search
    self.assertTrue(test_all_search((nr_points // 10), dim, bucket_size, query_radius))
AssertionError: False is not true
----------------------------------------------------------------------
```

Is this stochastic?

**Comment by Gasta88** [comment]
It appears so, I didn't notice that the value was floating between `nr_points` and `nr_points`-1.

I'll use a different test.

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**